File: skill-manager-app/src-tauri/src/models/skill.rs
use serde::Serialize;
use std::path::PathBuf;

#[derive(Debug, Clone, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct SkillFileEntry {
    pub relative_path: String,
    pub name: String,
    pub is_directory: bool,
    pub depth: usize,
}

#[derive(Debug, Clone, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct SkillFileContent {
    pub relative_path: String,
    pub content: String,
    pub language: String,
    pub is_markdown: bool,
    pub size: u64,
}

#[derive(Debug, Clone, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct AppOption {
    pub key: String,
    pub label: String,
}

#[derive(Debug, Clone, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct Skill {
    pub id: String,
    pub name: String,
    pub path: PathBuf,
    pub real_path: PathBuf,
    pub is_symlink: bool,
    pub is_broken_link: bool,
    pub description: String,
    pub version: String,
    pub author: Option<String>,
    pub category: Option<String>,
    pub custom_tags: Vec<String>,
    pub requires_agent: Option<String>,
    pub source_agents: Vec<String>,
    pub source_agent_labels: Vec<String>,
    pub body: String,
    pub warnings: Vec<String>,
}

File: skill-manager-app/src/types/skill.ts
export interface Skill {
  id: string
  name: string
  path: string
  realPath: string
  isSymlink: boolean
  isBrokenLink: boolean
  description: string
  version: string
  author?: string | null
  category?: string | null
  customTags: string[]
  requiresAgent?: string | null
  sourceAgents: string[]
  sourceAgentLabels: string[]
  body: string
  warnings: string[]
}

export interface AgentDir {
  key: string
  label: string
  path: string
  exists: boolean
  skillCount: number
}

export interface ScanIssue {
  path: string
  level: 'warning' | 'error'
  message: string
}

export interface ScanResult {
  skills: Skill[]
  agents: AgentDir[]
  issues: ScanIssue[]
  durationMs: number
}

export interface SkillFileEntry {
  relativePath: string
  name: string
  isDirectory: boolean
  depth: number
}

export interface SkillFileContent {
  relativePath: string
  content: string
  language: string
  isMarkdown: boolean
  size: number
}

export interface AppOption {
  key: string
  label: string
}

export type DetailViewMode = 'preview' | 'source'
export type SortMode = 'name' | 'agent' | 'version'
